packages/network/src/v2/schemas/decodeValue.ts

```diff
@@ -0,0 +1,110 @@
+import { SchemaType } from "@latticexyz/schema-type";
+import { decodeDynamicField } from "./decodeDynamicField";
+import { decodeStaticField } from "./decodeStaticField";
+
+// TODO: figure out how to use with SchemaTypeToPrimitive<T> return type to ensure correctness here
+export function decodeValue<T extends SchemaType>(schemaType: T, bytes: Uint8Array) {
+  switch (schemaType) {
+    case SchemaType.BOOL:
+    case SchemaType.UINT8:
+    case SchemaType.UINT16:
+    case SchemaType.UINT24:
+    case SchemaType.UINT32:
+    case SchemaType.UINT40:
+    case SchemaType.UINT48:
+    case SchemaType.UINT56:
+    case SchemaType.UINT64:
```

Comment:
For context - we need this instead of something like `getStaticByteLength(schemaType)` for proper types correct?

Reply:
yes, though this and the `decodeStaticField` method could probably be simplified with some `const` arrays instead of long case statements, but planning to come back to that as a "tidy" thing

Reply:
but also the types still don't exactly work due to some issues with mapping to primitives, this method still returns something like `string | number | bigint | boolean` for any `schemaType`